### PR TITLE
Lift test fixes

### LINKED ISSUE #644: Negative delta x causes sampling error
Default implementation uses Gamma distribution which doesn't work well with delta x (and delta y) being negative. 

Absolute value of the difference and delta_y could fix. Location of change: 
https://github.com/pymc-labs/pymc-marketing/blob/78964f0708179da63080a7d5c0398433616f774c/pymc_marketing/mmm/lift_test.py#L278-L287

Reproducible example:
```python
import numpy as np
import pandas as pd

from pymc_marketing.mmm import DelayedSaturatedMMM

data_url = "https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/data/mmm_example.csv"
data = pd.read_csv(data_url, parse_dates=["date_week"])

mmm = DelayedSaturatedMMM(
    date_column="date_week",
    channel_columns=["x1", "x2"],
    control_columns=[
        "event_1",
        "event_2",
        "t",
    ],
    adstock_max_lag=8,
    yearly_seasonality=2,
)

# Set features and target
X = data.drop("y", axis=1)
y = data["y"]

mmm.build_model(X, y)

# Delta x is negative 
df_lift_test = pd.DataFrame({
    "channel": ["x1"], 
    "x": [1], 
    "delta_x": [-0.1], 
    "delta_y": [-0.1], 
    "sigma": [0.05], 
})
mmm.add_lift_test_measurements(df_lift_test)
mmm.fit(X, y)
```

### PR DESCRIPTION
**Pull request by wd60622**
## Related Issue

- [x] Closes #644 
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [x] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--656.org.readthedocs.build/en/656/

### CODE REVIEW

File: pymc_marketing/mmm/lift_test.py [hunk 1]
```diff
@@ -63,8 +62,19 @@ def lift_test_indices(df_lift_test: pd.DataFrame, model: pm.Model) -> Indices:
     """
 
     columns = df_lift_test.columns.tolist()
+
+    def _ensure_datetime_type_in_array(x: np.ndarray) -> np.ndarray:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/lift_test.py:66]: can you add a basic test for this one ?

**Review by juanitorduz — CHANGES_REQUESTED** [review]
Looks great! Just a simple uni-test :)

File: pymc_marketing/mmm/lift_test.py (hunk 1 above)
Comment (ulfaslak) [inline pymc_marketing/mmm/lift_test.py:66]: I suggest changing the name to _coerce_datetype_type_in_array and being more explicit with types

```suggestion
    def _coerce_datetime_type_in_array(x: npt.NDArray[Any]) -> npt.NDArray[np.datetime64]:
```

File: pymc_marketing/mmm/lift_test.py [hunk 2]
```diff
@@ -63,8 +62,19 @@ def lift_test_indices(df_lift_test: pd.DataFrame, model: pm.Model) -> Indices:
...
+        if isinstance(first_value, pd.Timestamp):
+            x = np.array(x, dtype="datetime64[ns]")
+
+        return x
+
     return {
-        col: _lift_test_index(df_lift_test[col].to_numpy(), np.array(model.coords[col]))
+        col: _lift_test_index(
+            _ensure_datetime_type_in_array(df_lift_test[col].to_numpy()),
+            _ensure_datetime_type_in_array(np.array(model.coords[col])),
```
Comment (ulfaslak) [inline pymc_marketing/mmm/lift_test.py:76]:
```suggestion
            _coerce_datetime_type_in_array(df_lift_test[col].to_numpy()),
            _coerce_datetime_type_in_array(np.array(model.coords[col])),
```

**Review by ulfaslak — COMMENTED** [review]
🚀

File: pymc_marketing/mmm/lift_test.py (hunk 1 above)
Comment (wd60622) [reply pymc_marketing/mmm/lift_test.py]: Added tests for the dates.
Went for slightly different solution and added comment instead
Let me know what you think

Comment (wd60622) [reply pymc_marketing/mmm/lift_test.py]: Sorry. Missed this
Ended up changing the logic a bit and removed

File: pymc_marketing/mmm/lift_test.py [hunk 3]
```diff
@@ -63,8 +62,14 @@ def lift_test_indices(df_lift_test: pd.DataFrame, model: pm.Model) -> Indices:
     """
 
     columns = df_lift_test.columns.tolist()
+
     return {
-        col: _lift_test_index(df_lift_test[col].to_numpy(), np.array(model.coords[col]))
+        col: _lift_test_index(
+            df_lift_test[col].to_numpy(),
+            # Coords in the model become tuples of pd.Timestamp
+            # Convert to Series stores them as np.datetime64
```
Comment (wd60622) [inline pymc_marketing/mmm/lift_test.py:70]: Reference: https://github.com/pymc-devs/pymc/blob/04b6881efa9f69711d604d2234c5645304f63d28/pymc/model/core.py#L998

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
Thanks @wd60622 !

**Pull request merged**